### ISSUE TITLE
Kubuntu 22.04: castle-editor can't start: libgdk-x11-2.0.so.0 not found

### ISSUE THREAD

**Issue by LightTemplar** [castle-engine/castle-engine#392]
On my fresh installed Kubuntu 22.04 castle-editor didn't start. In Konsole there where message `error while loading shared libraries: libgdk-x11-2.0.so.0: cannot open shared object file: No such file or directory`

My workaround was to install `libgtk2.0-0`. Is it ok or should be used `libgtk-3-0`, which is installed with system?

**Comment by michaliskambi** [comment]
Indeed, CGE editor is using GTK 2 right now. That is because Lazarus LCL is using GTK 2 by default (on Linux and some similar platforms). Lazarus developers are working on GTK 3 widgetset, but for now it is "alpha" state (see https://wiki.freepascal.org/Widgetset ), not used by default (and we need at least 1 important fix to consider it for CGE editor -- TOpenGLControl with GTK 3 would have to support OpenGL context sharing).

I know this is a problem, with various distros moving away from GTK 2, and not having GTK 2 installed by default. But in this case we just have to wait for Lazarus LCL developers, and they already work on GTK 3.

Aside from it, we actually have to upgrade `TCastleWindow` backend (which is on our, CGE, side) from GTK 2 to GTK 3. This is just waiting for me (or any brave volunteer!) to have a bit of time and do it :)

I'm renaming the ticket -- we should hasten the GTK 2 -> GTK 3 upgrade, at least for our `TCastleWindow` backend, and maybe ping Lazarus LCL devs.

**Comment by michaliskambi** [comment]
I tested CGE editor with LCL GTK 3 widgetset now, but unfortunately many things are broken (attaching screenshots) - layout, image colors (red <-> blue), font sizes, any OpenGL context... So it really is "alpha", we have to wait in this regard for LCL devs.

![Zrzut ekranu z 2022-05-23 00-46-01](https://user-images.githubusercontent.com/1076978/169719259-c812956f-9d72-4ddf-ba20-ea765b000639.png)

![Zrzut ekranu z 2022-05-23 00-46-16](https://user-images.githubusercontent.com/1076978/169719262-e6be5b09-19cc-48fa-a3b1-fb9a865c1733.png)

**Comment by michaliskambi** [comment]
> who also want to use CGE in the GTK 3 distros.

Note that most distros are now "GTK 3 distros" in the sense that they have GTK 3 (or even 4) installed by default (as it is used by GNOME 3). And some of them don't have GTK 2 installed by default.

I don't think there should be anything special to do, except the things you already found out about :) One needs to install `libgtk2.0-dev` to have GTK 2 (both normal library and additional development files to develop applications that link to GTK 2, it will pull as dependencies the `libgtk2.0-0`). We mention it on https://castle-engine.io/compiling_from_source.php#section_Linux,_FreeBSD , though admittedly we should do better -- in the end, you have to install these libraries even if you don't compile CGE from sources.

Once we (both Lazarus LCL and TCastleWindow) migrate all to GTK 3, then you will have to install `libgtk-3-dev` . (Even on distros that ship with GTK 3, you will still usually need to install it, to develop applications that link with GTK 3.)

So, hopefully, the only thing that really changes is that
- now you should install `libgtk2.0-dev` package
- in the future you will have to install `libgtk-3-dev` package
- (and there's a possible transition period, when e.g. TCastleWindow will migrate to GTK 3 but Lazarus LCL will still use GTK 2, when you will need to install both `libgtk2.0-dev` and `libgtk-3-dev`)

Of course all testing and reports are absolutely appreciated, so keep it coming :) Thank you!
